### ISSUE TITLE
Infeasible OT problem while marginals sum to 1.

### ISSUE THREAD

**Issue by kilianFatras** [PythonOT/POT#342]
I am opening an issue because I found that in some examples, the OT problem is infeasible while the marginals sum to 1. I supsect this problems comes from different types of margianles (float32 vs float 64) as I explain at this end of this issue.

## Describe the bug
Infeasible OT problem while marginals sum to 1.

### To Reproduce
>>> import ot
>>> import torch
>>> import numpy as np
>>> X = torch.randn(5,10)
>>> Y = torch.randn(5,10)
>>> M = torch.cdist(X,Y)**2
>>> M.size()
torch.Size([5, 5])
>>> a = 1/5*torch.ones(5)
>>> b = ot.unif(5)
>>> a_np = a.numpy()
>>> ot.emd(a_np, b, M.numpy())
/home/fatras/.local/lib/python3.8/site-packages/ot/lp/__init__.py:276: UserWarning: Problem infeasible. Check that a and b are in the simplex
  result_code_string = check_result(result_code)
array([[0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.]])

### Expected behavior
It should return a transport plan.

### Environment (please complete the following information):
- Linux
- Python version: Python 3.8.10
- How was POT installed (source, `pip`, `conda`): pip

Output of the following code snippet:

>>> import platform; print(platform.platform())
Linux-5.13.0-27-generic-x86_64-with-glibc2.29
>>> import sys; print("Python", sys.version)
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0]
>>> import numpy; print("NumPy", numpy.__version__)
NumPy 1.19.4
>>> import scipy; print("SciPy", scipy.__version__)
SciPy 1.5.4
>>> import ot; print("POT", ot.__version__)
POT 0.7.0

### Additional context
Clearly I suspect that the problem is related to the float type of marginals. When I compute the command  "ot.emd(a, torch.from_numpy(b).float(), M.numpy())" it correctly returns me a transport plan. However the command "ot.emd(a.numpy().astype("float64"), b, M.numpy())" returns me the same infeasible message

**Comment by ncourty** [comment]
I am not sure what the expected output should be for a constant cost matrix, as all the transport plans will be minimum for this problem. While I agree that the error message is missleading, we  could simply output the matrix ```ab^T```. With a warning ?

**Comment by rflamary** [comment]
Actually by executing your example with the latest version of POT (0.8.1) I get a transport plan

```python
import ot
import torch
import numpy as np
X = torch.randn(5,10)
Y = torch.randn(5,10)
M = torch.cdist(X,Y)**2
M.size()
a = 1/5*torch.ones(5)
b = ot.unif(5)
a_np = a.numpy()
T=ot.emd(a_np, b, M.numpy())
```

with T equal to 

```
array([[0. , 0. , 0.2, 0. , 0. ],
       [0. , 0.2, 0. , 0. , 0. ],
       [0.2, 0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. , 0.2],
       [0. , 0. , 0. , 0.2, 0. ]], dtype=float32)
```
I think we did some cleanup on the type situation in ot.emd (everything goes to float64 in the c++ code anyways), could you confirm that the behavior is ok with you or give some example code that fails on the new version.

**Comment by kilianFatras** [comment]
Solved with 0.8.1 released.
